src/Common/CaresPTRResolver.cpp

```diff
@@ -0,0 +1,103 @@
+#include "CaresPTRResolver.h"
+#include <arpa/inet.h>
+#include <Common/Exception.h>
+#include "ares.h"
+#include "netdb.h"
+
+namespace DB {
+
+    namespace ErrorCodes
+    {
+        extern const int DNS_ERROR;
+    }
+
+    static void callback(void * arg, int status, int, struct hostent * host)
+    {
+        auto * ptr_records = reinterpret_cast<std::vector<std::string>*>(arg);
+        if (status == ARES_SUCCESS && host->h_aliases) {
+            int i = 0;
+            while (auto * ptr_record = host->h_aliases[i]) {
+                ptr_records->emplace_back(ptr_record);
+                i++;
+            }
+        }
```

Comment:
Because system `gethostbyaddr` may work this way and `ares_gethostbyaddr` has similar interface. But I'm not sure, maybe this situation is not possible with `c-ares` because of its implementation details.